tests/node.ts

```diff
@@ -0,0 +1,214 @@
+        console.log(`dialing ${cmdDef.targetIdentityName}`)
+        await node.dial(targetAddress)
+        console.log(`dialed`)
+        break
+      }
+      case 'msg': {
+        const targetPeerId = await peerIdForIdentity(cmdDef.targetIdentityName)
+        const relayPeerId = await peerIdForIdentity(cmdDef.relayIdentityName)
+        //@ts-ignore
+        let conn: Handler
+
+        console.log(`msg: dialing ${cmdDef.targetIdentityName} though relay ${cmdDef.relayIdentityName}`)
+        conn = await node.dialProtocol(
+          new Multiaddr(`/p2p/${relayPeerId}/p2p-circuit/p2p/${targetPeerId.toB58String()}`),
+          TEST_PROTOCOL
+        )
+        console.log(`piping msg: ${cmdDef.msg}`)
+
+        await pipe([new TextEncoder().encode(cmdDef.msg)], conn.stream, async (source: Stream['source']) => {
+          for await (const msg of source) {
+            const decoded = new TextDecoder().decode(msg.slice())
+
+            console.log(`Received <${decoded}>`)
+          }
+        })
+        console.log(`sent msg`)
+        break
+      }
+      default: {
+        throw new Error(`unknown cmd: ${cmdDef}`)
+      }
+    }
+  }
+}
+
+async function main() {
+  const argv = yargs(process.argv.slice(2))
+    .option('port', {
+      describe: 'node port',
+      type: 'number',
+      demandOption: true
+    })
+    .option('identityName', {
+      describe: 'node identity name',
+      choices: ['alice', 'bob', 'charly', 'dave', 'ed'],
+      demandOption: true
+    })
+    .option('bootstrapPort', {
+      describe: 'bootstrap node port',
+      type: 'number'
+    })
+    .option('bootstrapIdentityName', {
+      describe: 'identity name of a boostrap server',
+      choices: ['alice', 'bob', 'charly', 'dave', 'ed']
+    })
+    .option('noDirectConnections', {
+      type: 'boolean',
+      demandOption: true
+    })
+    .option('noWebRTCUpgrade', {
+      type: 'boolean',
+      demandOption: true
+    })
+    .option('command', {
+      describe: 'example: --command.name dial --command.targetIdentityName charly',
+      type: 'string'
+    })
+    .option('script', {
+      type: 'string',
+      demandOption: true
+    })
+    .coerce({
+      script: (input) => JSON.parse(input.replace(/'/g, '"'))
+    })
+    .parseSync()
+
+  let bootstrapAddress: Multiaddr | undefined
+
+  if (argv.bootstrapPort != null && argv.bootstrapIdentityName != null) {
+    const bootstrapPeerId = await peerIdForIdentity(argv.bootstrapIdentityName)
+    bootstrapAddress = new Multiaddr(`/ip4/127.0.0.1/tcp/${argv.bootstrapPort}/p2p/${bootstrapPeerId.toB58String()}`)
+  }
+  const peerId = await peerIdForIdentity(argv.identityName)
+
+  console.log(`running node ${argv.identityName} on port ${argv.port}`)
+  const node = await startNode({
+    peerId,
+    port: argv.port,
+    bootstrapAddress,
+    noDirectConnections: argv.noDirectConnections,
+    noWebRTCUpgrade: argv.noWebRTCUpgrade
+  })
+  handleProtocol(node)
```

Comment:
It could make sense to have this call before `node.start()` in `startNode()`, but probably not an issue